fundingmanager.go

```diff
@@ -1582,24 +1733,90 @@ func (f *fundingManager) sendFundingLocked(completeChan *channeldb.OpenChannel,
 	return nil
 }
 
-// sendChannelAnnouncement broadcast the necessary channel announcement
-// messages to the network. Should be called after the fundingLocked message
-// is sent (channelState is 'fundingLockedSent') and the channel is ready to
-// be used.
-func (f *fundingManager) sendChannelAnnouncement(completeChan *channeldb.OpenChannel,
+// addToRouterGraph sends a private ChannelAnnouncement and a private
+// ChannelUpdate to the gossiper so that it is added to the Router's internal
+// graph before the announcement_signatures is sent in
+// annAfterSixConfs. These private announcement messages are NOT
+// broadcasted to the greater network.
+func (f *fundingManager) addToRouterGraph(completeChan *channeldb.OpenChannel,
+	channel *lnwallet.LightningChannel,
+	shortChanID *lnwire.ShortChannelID, private bool) error {
+
+	chanID := lnwire.NewChanIDFromOutPoint(&completeChan.FundingOutpoint)
+
+	ann, err := f.newChanAnnouncement(f.cfg.IDKey, completeChan.IdentityPub,
+		channel.LocalFundingKey, channel.RemoteFundingKey, *shortChanID,
+		chanID)
+	if err != nil {
+		return fmt.Errorf("error generating channel "+
+			"announcement: %v", err)
+	}
+
+	// Send ChannelAnnouncement and ChannelUpdate to the gossiper to add
+	// to the Router's topology.
+	if err = f.cfg.SendLocalAnnouncement(ann.chanAnn); err != nil {
+		return fmt.Errorf("error sending private channel "+
+			"announcement: %v", err)
+	}
+	if err = f.cfg.SendLocalAnnouncement(ann.chanUpdateAnn); err != nil {
+		return fmt.Errorf("error sending private channel "+
+			"update: %v", err)
+	}
+
+	// As the channel is now added to the ChannelRouter's topology, the
+	// channel is moved to the next state of the state machine. It will be
+	// moved to the last state (actually deleted from the database) after
+	// the channel is finally announced.
+	err = f.saveChannelOpeningState(&completeChan.FundingOutpoint, addedToRouterGraph,
+		shortChanID)
+	if err != nil {
+		return fmt.Errorf("error setting channel state to"+
+			" addedToRouterGraph: %v", err)
+	}
+
+	return nil
+}
+
+// annAfterSixConfs broadcasts the necessary channel announcement messages to
+// the network after 6 confs. Should be called after the fundingLocked message
+// is sent and the channel is added to the router graph (channelState is
+// 'addedToRouterGraph') and the channel is ready to be used.
+func (f *fundingManager) annAfterSixConfs(completeChan *channeldb.OpenChannel,
 	channel *lnwallet.LightningChannel, shortChanID *lnwire.ShortChannelID) error {
 
-	// TODO(eugene) wait for 6 confirmations here
+	// Register with the ChainNotifier for a notification once the
+	// funding transaction reaches 6 confirmations.
+	txid := completeChan.FundingOutpoint.Hash
+	confNtfn, err := f.cfg.Notifier.RegisterConfirmationsNtfn(&txid, 6,
```

Comment:
I think this should be modified to actually announce after `max(6, completeChan.NumConfsRequired)`. The `FundingLocked` message is only sent once the channel has reached the required number of confirmations, and the channel announcement (if not private) should _only_ be sent after the `FundingLocked` message has been sent. As a result, there's a causal dependency that we need to ensure we uphold.